File: apps/api/src/modules/health/runtime-info.ts
const API_VERSION = '2.0.3';
const UNKNOWN_COMMIT = 'unknown';

export type RuntimeInfo = {
  service: 'evolua-api';
  version: string;
  commit: string;
  environment: string;
};

export function getRuntimeInfo(environment: string): RuntimeInfo {
  const renderCommit = process.env.RENDER_GIT_COMMIT?.trim();

  return {
    service: 'evolua-api',
    version: API_VERSION,
    commit: renderCommit ? renderCommit.slice(0, 12) : UNKNOWN_COMMIT,
    environment,
  };
}

File: apps/ai/app/runtime_info.py
import os
from importlib.metadata import PackageNotFoundError, version


def get_runtime_info(environment: str) -> dict[str, str]:
    try:
        service_version = version("evolua-ai")
    except PackageNotFoundError:
        service_version = "1.0.0"

    render_commit = os.getenv("RENDER_GIT_COMMIT", "").strip()
    return {
        "service": "evolua-ai",
        "version": service_version,
        "commit": render_commit[:12] if render_commit else "unknown",
        "environment": environment,
    }
